collect_app/src/test/java/org/odk/collect/android/activities/viewmodels/CurrentProjectViewModelTest.kt

```diff
@@ -0,0 +1,41 @@
+package org.odk.collect.android.activities.viewmodels
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.mock
+import org.odk.collect.android.projects.CurrentProjectProvider
+import org.odk.collect.androidshared.livedata.LiveDataUtils.getOrAwaitValue
+import org.odk.collect.projects.Project
+
+class CurrentProjectViewModelTest {
+    @get:Rule
+    var instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    private lateinit var currentProjectProvider: CurrentProjectProvider
+    private lateinit var currentProjectViewModel: CurrentProjectViewModel
+
+    @Before
+    fun setup() {
+        currentProjectProvider = mock(CurrentProjectProvider::class.java)
+        `when`(currentProjectProvider.getCurrentProject()).thenReturn(Project("Project X", "X", "#cccccc"))
+        currentProjectViewModel = CurrentProjectViewModel(currentProjectProvider)
+    }
+
+    @Test
+    fun `Initial current project should be set`() {
+        assertThat(currentProjectViewModel.currentProject.value!!.value, `is`(Project("Project X", "X", "#cccccc")))
+    }
+
+    @Test
+    fun `setCurrentProject() should add new project to live data`() {
+        val project = Project("Project Y", "Y", "#ffffff")
+
+        currentProjectViewModel.setCurrentProject(project)
+        assertThat(getOrAwaitValue(currentProjectViewModel.currentProject).value, `is`(project))
```

Comment:
I don't think you need `getOrAwaitValue` here. I gave it a try and the test worked without it. The `InstanceTaskExecutor` or just using Robolectric should take care of the weirdness in `LiveData` (although the former is better as the test is faster).

Reply:
Are you sure? I've just tried and it doesn't work without that utility method

Reply:
Yeah if I remove `getOrAwait` so it's just:

```kotlin
assertThat(currentProjectViewModel.currentProject.value!!.value, `is`(project))
```

it works.

Reply:
Hmm magic I tried the same and it failed now it works... nevermind. Fixed.